code/obj/critter/misc.dm

```diff
@@ -487,22 +489,42 @@
 			else
 				continue
 
+	attackby(obj/item/W as obj, mob/M as mob)
+		if(istype(W, /obj/item/reagent_containers/food/snacks) && !(M in src.friends))
+			if(prob(20))
+				src.visible_message("<span class='notice'>[src] chitters happily at the [W], it will no longer attack [M]!</span>")
+				friends += M
+				playsound(src.loc, "sound/misc/bugchitter.ogg", 50, 0)
+				src.task = "thinking"
+			else
+				src.visible_message("<span class='notice'>[src] hated the [W]! It bit [M]'s hand!</span>")
+				random_brute_damage(M, rand(6,12),1)
+				playsound(src.loc, "sound/items/Wirecutter.ogg", 50, 0)
+				M.emote("scream")
+			M.drop_item()
+			qdel(W)
+			src.health = min(src.maxhealth, src.health + health_gain_from_food)
+			eat_twitch(src)
+		else
+			..()
+
+
 	ChaseAttack(mob/M)
 		..()
-		if(prob(50))
-			if (M.reagents)
+		if(!ON_COOLDOWN(src, "scorpion_ability", 10 SECONDS))
+			if(prob(50))
 				M.visible_message("<span class='combat'><B>[src]</B> stings [src.target]!</span>")
-				M.reagents.add_reagent("neurotoxin", 15)
-				M.reagents.add_reagent("toxin", 6)
+				M.reagents?.add_reagent("neurotoxin", 15)
+				M.reagents?.add_reagent("toxin", 6)
 				playsound(src.loc, "sound/impact_sounds/Generic_Stab_1.ogg", 50, 1)
 				M.emote("scream")
-				M.add_karma(1)
-		else
-			random_brute_damage(M, rand(5,10),1)
-			M.visible_message("<span class='combat'><B>[src]</B> tries to grab [src.target] with its pincers!</span>")
-			playsound(src.loc, "sound/items/Wirecutter.ogg", 50, 0)
-			M.changeStatus("weakened", 4 SECONDS)
-			M.force_laydown_standup()
+			else
+				random_brute_damage(M, rand(5,10),1)
+				M.visible_message("<span class='combat'><B>[src]</B> tries to grab [src.target] with its pincers!</span>")
```

Comment:
considering it doesn't actually grab you in the way that the trench fish do, "tackles" may be a better suited word to what it does

Reply:
sounds pretty good!